image-scanner.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+# Container scanning is the process of analyzing components within containers to uncover 
+# potential security threats. It is integral to ensuring that the software remains secure 
+# as it progresses through the application life cycle. Container scanning takes its cues 
+# from practices like vulnerability scanning and penetration testing.
+
+# This script uses Trivy, a simple and comprehensive vulnerability scanner for containers
+# and other artifacts. Trivy can find vulnerabilities, IaC misconfigurations, secrets, 
+# SBOM discovery, Cloud scanning, Kubernetes security risks, and more. 
+# More information about Trivy can be found at https://trivy.dev/.
+
+# This shell script automates the process of scanning container images from a specific Docker 
+# organization (passed in, as an argument) using the Trivy vulnerability scanner. The script 
+# retrieves a list of Docker repositories from Docker Hub and runs Trivy scans on each image 
+# to generate security reports. These reports are saved as HTML files in a designated directory.
+
+# Check if the organization name is provided as an argument
+if [ -z "$1" ]; then
+    echo "Error: Please provide the Docker organization name as an argument."
+    echo "Usage: $0 <organization_name>"
+    exit 1
+fi
+
+# Get Docker Organization Name from the script argument
+ORG="$1"
+
+# Check if trivy is installed and install it if not
+if ! command -v trivy &> /dev/null; then
+    echo "Trivy is not installed. Installing..."
+    wget https://github.com/aquasecurity/trivy/releases/download/v0.51.1/trivy_0.51.1_Linux-64bit.deb
+    sudo dpkg -i trivy_0.51.1_Linux-64bit.deb
+else
+    echo "Found trivy, using trivy v$(trivy -v | cut -d ' ' -f 2)"
+fi
+
+echo "Retrieving repository list ..."
+REPO_LIST=$(curl -s "https://hub.docker.com/v2/repositories/${ORG}/?page_size=100" | jq -r '.results|.[]|.name')
+
+# Get today's date in the desired format for folder naming
+TODAY_DATE=$(date +'%d-%m-%Y')
+# Define the root and sub directory name
+ROOT_DIR="image-scanner-reports"
+DIR="bahmni-${TODAY_DATE}"
+
+# Check if the root directory exists and create it if not
+if [ ! -d "$ROOT_DIR" ]; then
+    echo "Creating directory $ROOT_DIR"
+    mkdir "$ROOT_DIR"
+else
+    echo "Root Directory $ROOT_DIR already exists, proceeding"
+fi
+
+# Check if the root directory exists and create it if not
```

Comment:
@rahu1ramesh This is duplicated. Please remove

Reply:
@mohan-13 This checks for the subdirectory, which can be reused but may not have significant impact.